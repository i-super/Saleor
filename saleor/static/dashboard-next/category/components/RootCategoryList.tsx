import AddIcon from "material-ui-icons/Add";
import Card from "material-ui/Card";
import IconButton from "material-ui/IconButton";
import * as React from "react";

import CategoryList from "../../components/CategoryList";
import PageHeader from "../../components/PageHeader";
import { RootCategoryChildrenQuery } from "../../gql-types";
import i18n from "../../i18n";

interface RootCategoryListProps {
  data: RootCategoryChildrenQuery;
  loading?: boolean;
  onClick?(id: string);
  onCreate?();
}

const RootCategoryList: React.StatelessComponent<RootCategoryListProps> = ({
  data,
  loading,
  onClick,
  onCreate
}) => (
  <Card>
    <PageHeader title={i18n.t("Categories", { context: "title" })}>
      <IconButton onClick={onCreate} disabled={loading}>
        <AddIcon />
      </IconButton>
    </PageHeader>
    <CategoryList
      categories={data.categories && data.categories.edges}
      onClick={onClick}
    />
  </Card>
);

export default RootCategoryList;
