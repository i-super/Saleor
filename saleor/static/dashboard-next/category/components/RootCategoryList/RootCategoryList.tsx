import AddIcon from "material-ui-icons/Add";
import Card from "material-ui/Card";
import IconButton from "material-ui/IconButton";
import * as React from "react";

import CategoryList from "../../../components/CategoryList";
import PageHeader from "../../../components/PageHeader";
import { RootCategoryChildrenQuery } from "../../../gql-types";
import i18n from "../../../i18n";

interface RootCategoryListProps {
  categories?: Array<{
    id: string;
    name: string;
  }>;
  onClick?(id: string);
  onCreate?();
}

const RootCategoryList: React.StatelessComponent<RootCategoryListProps> = ({
  categories,
  onClick,
  onCreate
}) => (
  <Card>
    <PageHeader title={i18n.t("Categories", { context: "title" })}>
      <IconButton onClick={onCreate}>
        <AddIcon />
      </IconButton>
    </PageHeader>
    <CategoryList categories={categories} onClick={onClick} />
  </Card>
);

export default RootCategoryList;
