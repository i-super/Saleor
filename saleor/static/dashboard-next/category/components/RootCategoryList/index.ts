export { default } from "./RootCategoryList";
export * from "./RootCategoryList";
