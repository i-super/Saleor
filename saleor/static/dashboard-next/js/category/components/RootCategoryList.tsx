import AddIcon from "material-ui-icons/Add";
import Card from "material-ui/Card";
import IconButton from "material-ui/IconButton";
import * as React from "react";

import PageHeader from "../../components/PageHeader";
import i18n from "../../i18n";
import { RootCategoryChildrenQuery } from "../gql-types";
import CategoryList from "./CategoryList";

interface RootCategoryListProps {
  data: RootCategoryChildrenQuery;
  loading?: boolean;
  onClick?(id: string);
  onCreate?();
}

const RootCategoryList: React.StatelessComponent<RootCategoryListProps> = ({
  data,
  loading,
  onClick,
  onCreate
}) => (
  <Card>
    <PageHeader title={i18n.t("Categories", { context: "title" })}>
      <IconButton onClick={onCreate} disabled={loading}>
        <AddIcon />
      </IconButton>
    </PageHeader>
    <CategoryList
      categories={data.categories && data.categories.edges}
      onClick={onClick}
    />
  </Card>
);

export default RootCategoryList;
