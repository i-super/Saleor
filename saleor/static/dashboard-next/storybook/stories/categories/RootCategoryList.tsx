import { storiesOf } from "@storybook/react";
import { DialogContentText } from "material-ui/Dialog";
import * as React from "react";

import RootCategoryList from "../../../category/components/RootCategoryList";

const categories = [
  {
    id: "123123",
    name: "Lorem ipsum dolor"
  },
  {
    id: "876752",
    name: "Mauris vehicula tortor vulputate"
  }
];

storiesOf("Categories / RootCategoryList", module)
  .add("with initial data", () => (
    <RootCategoryList
      categories={categories}
      onClick={() => {}}
      onCreate={() => {}}
    />
  ))
  .add("without initial data", () => (
    <RootCategoryList categories={[]} onClick={() => {}} onCreate={() => {}} />
  ))
  .add("when loading data", () => (
    <RootCategoryList onClick={() => {}} onCreate={() => {}} />
  ));
