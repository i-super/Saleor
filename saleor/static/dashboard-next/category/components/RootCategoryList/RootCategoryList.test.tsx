import * as React from "react";
import { MemoryRouter } from "react-router-dom";
import * as renderer from "react-test-renderer";

import RootCategoryList from "./";

describe("<CategorySubcategories />", () => {
  it("renders with initial data", () => {
    const categories = [
      {
        id: "123123",
        name: "Lorem ipsum dolor"
      },
      {
        id: "876752",
        name: "Mauris vehicula tortor vulputate"
      }
    ];
    const component = renderer.create(
      <RootCategoryList
        categories={categories}
        onClick={jest.fn()}
        onCreate={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders without initial data", () => {
    const component = renderer.create(
      <RootCategoryList
        categories={[]}
        onClick={jest.fn()}
        onCreate={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders when loading data", () => {
    const component = renderer.create(
      <RootCategoryList onClick={jest.fn()} onCreate={jest.fn()} />
    );
    expect(component).toMatchSnapshot();
  });
});
