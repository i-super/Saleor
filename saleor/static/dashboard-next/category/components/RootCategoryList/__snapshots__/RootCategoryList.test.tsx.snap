// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategorySubcategories /> renders when loading data 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-elevation2-6 MuiPaper-rounded-3 MuiCard-root-1"
>
  <div
    className="Component-root-32"
  >
    <h2
      className="MuiTypography-root-35 MuiTypography-title-41 Component-title-34"
    >
      Categories
    </h2>
    <div
      className="Component-action-29"
    >
      <button
        className="MuiButtonBase-root-65 MuiIconButton-root-59"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-64"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-68"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-74"
        />
      </button>
    </div>
  </div>
  <ul
    className="MuiList-root-81 MuiList-padding-82"
  >
    <li
      className="MuiListItem-root-85 MuiListItem-default-88 MuiListItem-gutters-92"
      disabled={false}
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-68 MuiListItemIcon-root-95"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
          />
        </g>
      </svg>
      <div
        className="MuiListItemText-root-96"
      >
        <h3
          className="MuiTypography-root-35 MuiTypography-subheading-42 MuiListItemText-primary-99"
        >
          <span
            className="Component-skeleton-107"
            style={undefined}
          >
            â€Œ
          </span>
        </h3>
      </div>
    </li>
  </ul>
</div>
`;

exports[`<CategorySubcategories /> renders with initial data 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-elevation2-6 MuiPaper-rounded-3 MuiCard-root-1"
>
  <div
    className="Component-root-32"
  >
    <h2
      className="MuiTypography-root-35 MuiTypography-title-41 Component-title-34"
    >
      Categories
    </h2>
    <div
      className="Component-action-29"
    >
      <button
        className="MuiButtonBase-root-65 MuiIconButton-root-59"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-64"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-68"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-74"
        />
      </button>
    </div>
  </div>
  <ul
    className="MuiList-root-81 MuiList-padding-82"
  >
    <div
      className="MuiButtonBase-root-65 MuiListItem-root-85 MuiListItem-default-88 MuiListItem-gutters-92 MuiListItem-button-93"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-68 MuiListItemIcon-root-95"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
          />
        </g>
      </svg>
      <div
        className="MuiListItemText-root-96"
      >
        <h3
          className="MuiTypography-root-35 MuiTypography-subheading-42 MuiListItemText-primary-99"
        >
          Lorem ipsum dolor
        </h3>
      </div>
      <span
        className="MuiTouchRipple-root-74"
      />
    </div>
    <div
      className="MuiButtonBase-root-65 MuiListItem-root-85 MuiListItem-default-88 MuiListItem-gutters-92 MuiListItem-button-93"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-68 MuiListItemIcon-root-95"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
          />
        </g>
      </svg>
      <div
        className="MuiListItemText-root-96"
      >
        <h3
          className="MuiTypography-root-35 MuiTypography-subheading-42 MuiListItemText-primary-99"
        >
          Mauris vehicula tortor vulputate
        </h3>
      </div>
      <span
        className="MuiTouchRipple-root-74"
      />
    </div>
  </ul>
</div>
`;

exports[`<CategorySubcategories /> renders without initial data 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-elevation2-6 MuiPaper-rounded-3 MuiCard-root-1"
>
  <div
    className="Component-root-32"
  >
    <h2
      className="MuiTypography-root-35 MuiTypography-title-41 Component-title-34"
    >
      Categories
    </h2>
    <div
      className="Component-action-29"
    >
      <button
        className="MuiButtonBase-root-65 MuiIconButton-root-59"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-64"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-68"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-74"
        />
      </button>
    </div>
  </div>
  <ul
    className="MuiList-root-81 MuiList-padding-82"
  >
    <li
      className="MuiListSubheader-root-102 MuiListSubheader-sticky-106"
    >
      No categories found
    </li>
  </ul>
</div>
`;
